src/gs-page.c

```diff
@@ -481,6 +515,45 @@ gs_page_install_app (GsPage *page,
 	}
 }
 
+void
+gs_page_copy_app (GsPage *page,
+		  GsApp *app,
+		  const gchar *dest,
+		  GsShellInteraction interaction,
+		  GCancellable *cancellable)
+{
+	GsPagePrivate *priv = gs_page_get_instance_private (page);
+	GsPageHelper *helper;
+
+	helper = g_slice_new0 (GsPageHelper);
```

Comment:
Nitpick: You can initialize it together with the declaration. And use `g_new0` is probably better too.

Reply:
I've merged the lines locally.

But the free function uses `g_slice_free()` and all the other instances of `GsPageHelper` allocate it with `g_slice_new0()` so we'd need to change those as well and I don't see a critical need to do that.

Why do you prefer `g_new0()`? Because it's less of a risk when debugging and the performance gains of `g_slice` allocation are trivial here?

Reply:
I read somewhere that `g_slice` was no longer a faster solution. But I hadn't noticed the free func, so let's keep it then.